examples/gaussian_processes/HSGP-Advanced.myst.md

```diff
@@ -0,0 +1,896 @@
+1. It **can only be used with _stationary_ covariance kernels** such as the Matern family.  The {class}`~pymc.gp.HSGP` class is compatible with any `Covariance` class that implements the `power_spectral_density` method.  There is a special case made for the `Periodic` covariance, which is implemented in PyMC by The {class}`~pymc.gp.HSGPPeriodic`.
+2. It **does not scale well with the input dimension**.  The HSGP approximation is a good choice if your GP is over a one dimensional process like a time series, or a two dimensional spatial point process.  It's likely not an efficient choice where the input dimension is larger than three. 
+3. It **_may_ struggle with more rapidly varying processes**.  If the process you're trying to model changes very quickly relative to the extent of the domain, the HSGP approximation may fail to accurately represent it.  We'll show in later sections how to set the accuracy of the approximation, which involves a trade-off between the fidelity of the approximation and the computational complexity.
+4. **For smaller data sets, the full unapproximated GP may still be more efficient**.
+
+A secondary goal of this implementation is flexibility via an accessible implementation where the core computations are implemented in a modular way.  For basic usage, users can use the `.prior` and `.conditional` methods and essentially treat the HSGP class as a drop in replacement for `pm.gp.Latent`, the unapproximated GP.  More advanced users can bypass those methods and work with `.prior_linearized` instead, which exposes the HSGP as a parametric model.  For more complex models with multiple HSGPs, users can work directly with functions like `pm.gp.hsgp_approx.calc_eigenvalues` and `pm.gp.hsgp_approx.calc_eigenvectors`.
+
+#### References:
+- Original reference: [Solin & Sarkka, 2019](https://link.springer.com/article/10.1007/s11222-019-09886-w).
+- HSGPs in probabilistic programming languages: [Riutort-Mayol et al., 2020](https://arxiv.org/abs/2004.11408).
+- HSTPs (Student-t process): [Sellier & Dellaportas, 2023](https://proceedings.mlr.press/v202/sellier23a.html).
+- Kronecker HSGPs: [Dan et al., 2022](https://arxiv.org/pdf/2210.11358.pdf)
+- PyMC's {class}`~pymc.gp.HSGP` API
+
+```{code-cell} ipython3
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import pymc as pm
+import pytensor.tensor as pt
+```
+
+```{code-cell} ipython3
+az.style.use("arviz-whitegrid")
+plt.rcParams["figure.figsize"] = [12, 5]
+%config InlineBackend.figure_format = 'retina'
+seed = sum(map(ord, "hsgp advanced"))
+rng = np.random.default_rng(seed)
+```
+
+# Example 1: A hierarchical HSGP, a more custom model
+
++++
+
+`````{admonition} Looking for a beginner's introduction?
+:class: tip
+This notebook is the second part of our HSGP tutorials. We strongly recommend you start by reading [the first part](HSGP-Basic.myst.md), which is a smoother introduction to HSGPs and covers more basic use-cases.
```

Comment:
This is not generating a cross-reference. This is a reference with explicit text to a target within the same project, so it should use one of the two options in the first code cell of https://github.com/pymc-devs/pymc/pull/7235#issuecomment-2032750262. That is, instead of `[the first part](HSGP-Basic.myst.md)`:

```
{ref}`the first part <hsgp>`
[the first part](#hsgp)
```